src/inbox/mod_inbox_muc.erl

```diff
@@ -0,0 +1,100 @@
+%%%-------------------------------------------------------------------
+%%% @author Dominik Stanaszek dominik.stanaszek@erlang-solutions.com
+%%% @copyright (C) 2018, Erlang-Solutions
+%%% @doc
+%%%
+%%% @end
+%%% Created : 6.07.2018 
+%%%-------------------------------------------------------------------
+-module(mod_inbox_muc).
+-author("dominik.stanaszek@erlang-solutions.com").
+-include("jlib.hrl").
+
+-export([update_inbox/5, start/1, stop/1]).
+
+start(Host) ->
+    ejabberd_hooks:add(update_inbox, Host, ?MODULE, update_inbox, 90),
+    % TODO check ooptions: if system messages stored -> 
+    % add hook handler for system messages on hook ie. invitation_sent
+    ok.
+
+stop(Host) ->
+    ejabberd_hooks:delete(update_inbox, Host, ?MODULE, update_inbox, 90),
+    ok.
+
+update_inbox(Acc, Room, {From, FromRoomJid}, AffsDict, Packet) ->
+    Affs = dict:to_list(AffsDict),
+    Users = affs_to_allowed_users(Affs),
+    FromBare = jid:to_bare(From),
+    lists:foreach(fun(User) -> update_inbox(Room, 
+                                            {FromBare, FromRoomJid}, 
+                                            User, 
+                                            Packet) end,
+                  Users),
+    Acc.
+
+affs_to_allowed_users(Users) ->
+    AllowedUsers = lists:filter(fun({_User, outcast}) -> false;
+                    (_) -> true end, Users),
+    lists:map(fun({{User, Domain, Res}, _Aff}) -> jid:make(User, Domain, Res) end,
+              AllowedUsers).
+
+
+update_inbox(Room, {From, FromRoomJid}, To, Packet) ->
+    Host = To#jid.server,
+    case direction(From, To) of
+        outgoing ->
+            NewPacket = alter_senders_packet(Packet, FromRoomJid, To),
+            handle_outgoing_message(Host, From, Room, NewPacket);
+        incoming ->
+            NewPacket = alter_receivers_packet(Packet, FromRoomJid, To),
+            handle_incoming_message(Host, Room, To, NewPacket)
+    end.
+
+direction(From, To) ->
+    case jid:are_equal(From, To) of
+        true -> outgoing;
+        false -> incoming
+    end.
+
+
+% TODO this logic probably should be in mod_muc_room before routing
```

Comment:
hmm, it was working before without "from" and "to" attributes.
If inbox need them, then it can add them to its copy of message.

btw, there is `jlib:replace_from_to/3`, which can be used here.

Reply:
It's not that inbox needs them, I just wanted to have the same message stanza in inbox as the one user sends/receives.